File: BowlPicks.Api/Models/UserPicksContainerModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using BowlPicks.Api.Entity.Model;

namespace BowlPicks.Api.Models
{
    public class UserPicksContainerModel
    {
        public UserPicksContainerModel(IEnumerable<UserPick> picks)
        {
            var pickModels = new List<UserPickModel>();

            foreach (var pick in picks)
            {
                pickModels.Add(new UserPickModel(pick));

                if (pick.Game.IsGameOver)
                {
                    ++TotalPicks;

                    if ((pick.IsTeam1Selected && pick.Game.DidTeam1Win) ||
                        (!pick.IsTeam1Selected && !pick.Game.DidTeam1Win))
                    {
                        MaxPoints += pick.Confidence;
                        ++CorrectPicks;
                        
                        Points += pick.Confidence;
                    }
                }
                else
                {
                    MaxPoints += pick.Confidence;
                }
            }

            PickPercent = TotalPicks==0?0:(decimal)CorrectPicks / TotalPicks;

            //TODO Set Rank
            Rank = 1;

            var firstPick = picks.FirstOrDefault();

            if (firstPick != null)
            {
                var user = firstPick.User;

                Name = user.Username;
                UserId = user.UserId;
            }

            Picks = pickModels;
        }

        public IEnumerable<UserPickModel> Picks { get; set; }
        public string Name { get; set; }
        public int CorrectPicks { get; set; }
        public int TotalPicks { get; set; }
        public decimal PickPercent { get; set; }
        public int Points { get; set; }
        public int MaxPoints { get; set; }
        public int Rank { get; set; }
        public int UserId { get; set; }

        public LeaderboardModel GetLeaderboardModel()
        {
            return new LeaderboardModel
            {
                CorrectPicks = CorrectPicks,
                MaxPoints = MaxPoints,
                Name = Name,
                PickPercent = PickPercent,
                Points = Points,
                Rank = Rank,
                TotalPicks = TotalPicks
            };
        }
    }
}

File: BowlPicks/app/models/UserPicksContainerModel.js
"use strict";

if (typeof BowlPicks == "undefined") {
    window.BowlPicks = {};
}

if (typeof BowlPicks.Models == "undefined") {
    BowlPicks.Models = {};
}

if (typeof BowlPicks.Models.UserPicksContainerModel == "undefined") {
    BowlPicks.Models.UserPicksContainerModel = function () {
        var self = this;

        this.Picks = new Array();
        this.Name = null;
        this.CorrectPicks = 0;
        this.TotalPicks = 0;
        this.PickPercent = 0.0;
        this.Points = 0;
        this.MaxPoints = 0;
        this.Rank = 0;
        this.IsHeader = false;
        this.UserId = 0;
        this.PicksFinal = true;

        this.Init = function (data) {
            self.Name = data.Name;
            self.CorrectPicks = data.CorrectPicks;
            self.TotalPicks = data.TotalPicks;
            self.PickPercent = data.PickPercent;
            self.Points = data.Points;
            self.MaxPoints = data.MaxPoints;
            self.Rank = data.Rank;
            self.UserId = data.UserId;
            self.PicksFinal = data.PicksFinal;

            while (self.Picks.length > 0) {
                self.Picks.pop();
            }

            for (var i = 0; i < data.Picks.length; i++) {
                var picksModel = new BowlPicks.Models.UserPickModel();
                picksModel.Init(data.Picks[i]);
                self.Picks.push(picksModel);
            }

        };
    };
}